azure-mgmt-compute/src/main/java/com/microsoft/azure/management/compute/VirtualMachine.java

```diff
@@ -98,52 +99,235 @@
+
+        /**
+         * Specifies the specialized Linux OS disk to be attached to the virtual machine.
+         * @param osDiskUrl The url to the OS disk in the Azure Storage account.
+         * @return The next stage of the Linux virtual machine definition.
+         */
+        DefinitionWithDataDisk withLinxOSDisk(String osDiskUrl);
     }
 
-    interface DefinitionWithOSDisk extends DefinitionWithVMImage {
-        OSDiskFromImage defineOSDisk(String name);
-        DefinitionWithDataDisk withUserImage(String containerName, String vhdName, OperatingSystemTypes osType);
-        DefinitionWithDataDisk attachOSDisk(String containerName, String vhdName, OperatingSystemTypes osType);
+
+    interface DefinitionWithOSType {
+        /**
+         * Specifies the OS type of the virtual machine as Linux.
+         * @return The next stage of the Windows virtual machine definition.
+         */
+        DefinitionWithRootUserName withLinuxOS();
+
+        /**
+         * Specifies the OS type as Windows.
+         * @return The next stage allowing optionally setting the Windows configuration.
+         */
+        DefinitionWithOptionalWindowsConfiguration withWindowsOS();
     }
 
-    interface NewOSDiskFromImage {
-        DefinitionWithOSDiskConfiguration fromImage(ImageReference imageReference);
-        DefinitionWithOSDiskConfiguration fromLatestImage(String publisher, String offer, String sku);
-        DefinitionWithOSDiskConfiguration fromKnownImage(KnownVirtualMachineImage knownImage);
+    interface DefinitionWithOptionalWindowsConfiguration {
+        /**
+         * Specifies the configuration for the Windows virtual machine.
+         * @return The next stage for setting the configuration.
+         */
+        DefinitionWithWindowsConfiguration defineConfiguration();
+
+        /**
+         * Specifies the administrator user name for the Windows virtual machine.
+         * @param userName The Windows administrator user name to use. This must follow the required naming convention for Windows user name.
+         * @return The next stage of the Windows virtual machine definition.
+         */
+        DefinitionWithPassword withAdminUserName(String userName);
     }
 
-    interface OSDiskFromImage extends NewOSDiskFromImage {
-        DefinitionWithOSDiskConfiguration fromUserImage(String containerName, String vhdName, OperatingSystemTypes osType);
-        DefinitionWithCommonOSDiskConfiguration useDisk(String containerName, String vhdName, OperatingSystemTypes osType);
+    interface DefinitionWithWindowsConfiguration {
+        /**
+         * Specifies that VM Agent should be provisioned.
+         * @return The optional windows configurations
+         */
+        DefinitionWithWindowsConfiguration enableVMAgent();
+
+        /**
+         * Specifies that VM Agent should not be provisioned.
+         * @return The optional windows configurations
+         */
+        DefinitionWithWindowsConfiguration disableVMAgent();
```

Comment:
I see, then maybe we can remove `enable`s. As Windows recommends turning on AutoUpdate as well.